cameraServer.py

```diff
@@ -55,56 +58,106 @@ def startCamera(config):
     return cs, camera
 
 
-#Process Functions
-def getRetroPos(img, display=False, distance_away=distance_away):
-    """Function for finding retro-reflective tape"""
-
-    hsv = cv2.cvtColor(img, cv2.COLOR_BGR2HSV) #Convert to HSV to make the mask easier
-    mask = cv2.inRange(hsv, lowerGreen, higherGreen) #Create a mask of everything in between the greens
-    mask = cv2.dilate(mask, None, iterations=1) #Expand the mask to allow for further away tape
-
-    contours = cv2.findContours(mask, 1, 2)[1] #Find the contours
-    
-    if len(contours) > 1: #Get contours with area above magic number 10 and append its smallest rectangle
-        rects = []
-        for cnt in contours:
-            if cv2.contourArea(cnt) > minContourArea:
-                rects.append(cv2.minAreaRect(cnt))
-    
-        pairs = []
-        leftRect = None
-        for rect in sorted(rects, key=lambda x:x[0]): #Get rectangle pairs
-            if (leftAngleSize - angleOffset) < rect[2] < (leftAngleSize + angleOffset):
-                leftRect = rect
-            elif (rightAngleSize - angleOffset) < rect[2] < (rightAngleSize + angleOffset):
-                if leftRect:
-                    pairs.append((leftRect, rect))
-                    leftRect = None
-                    
-        if len(pairs) >= 1:
-            closestToMiddle = min(pairs, key = lambda x:abs((x[0][0][0]+x[1][0][0]) - screenSize[0]))
+# Process Functions
+def getDistance(boxes):
+    if boxes is None:
+        return math.nan, math.nan
+    Lpoint = max(boxes[0], key=lambda x: x[0])
+    Rpoint = min(boxes[1], key=lambda x: x[0])
+    width = abs(Lpoint[0] - Rpoint[0])
+    mid = (Rpoint[0] + Lpoint[0]) / 2
+    distance_from_center = mid - screenX / 2
+    offset = getOffset(width, distance_from_center)
+    if width > 0:
+        dist = (realTapeDistance * focal_length) / width
+        return dist, offset
+    else:
+        return math.nan, offset
+
+
+def getOffset(width, x):
+    # if width = 20cm then what is x in cm
+    offset = x / (width / (realTapeDistance))
+    return -offset
+
+
+def createAnnotatedDisplay(
+    frame: np.array, pairs: list, closestToMiddle: tuple, circle: tuple
+):
+    img = cv2.line(frame, (160, 0), (160, 240), (255, 0, 0), thickness=1)
+    for pair in pairs:
+        if (pair[0][1][0] == closestToMiddle[0][0]).all():
+            colour = (0, 255, 0)
```

Comment:
It would be nicer to keep these BGR values in a variable so that people know what colour your using.